[Jessica Mary] iP

maryjess opened the pull request:
### Duke
> “Each day I will accomplish one thing on my to-do list.” ― Lailah Gifty Akita

Duke is a **___chatbot application___** that helps to manage:
* Deadlines
* Events
* To-dos

It is super simple to use. All you need to do:
1. Download Duke [here](https://github.com/maryjess/ip/releases/tag/A-Jar)
2. Double-click and run the .jar file
3. Duke is ready to manage your tasks! ☺ 

Features of Duke v0.1:
- [X] Add task
- [X] Mark task
- [X] Unmark task
- [X] Delete task
- [X] Show task list
- [ ] Search task (coming soon)

Duke is written in Java. Here is the `Duke` method:
```java
public class Duke {
    public static void main(String[] args) {
        String home = System.getProperty("user.home");
        Path path = Paths.get(home, "data", "duke");
        new Duke(path).run();
    }
}
```
Check out its [source code](https://github.com/maryjess/ip) for programmers learning Java 😃

maryjess committed:
Create Level-1

maryjess committed:
Tidy up code

maryjess committed:
Create Level-2

maryjess committed:
Create Level-3

maryjess committed:
Create Level-4

maryjess committed:
Create automated text UI testing

maryjess committed:
Create Level-5

maryjess committed:
Additional error handling

maryjess committed:
Create Level-6

maryjess committed:
Create Level-7

maryjess committed:
Tidy up code

maryjess committed:
Create Level-8

maryjess committed:
Add get task at specific date

maryjess committed:
Merge branch 'branch-Level-7'

maryjess committed:
Merge branch 'branch-Level-8'

maryjess committed:
Handle minor error

maryjess committed:
Create A-MoreOOP

maryjess committed:
Add package

maryjess committed:
Set up gradle

maryjess committed:
Create JUnit tests

maryjess committed:
Create JAR file

maryjess committed:
Create JavaDoc comments

maryjess committed:
Create Level-9

maryjess committed:
Merge branch 'branch-Level-9'

maryjess committed:
Merge branch 'branch-A-JavaDoc'

maryjess committed:
Follow Coding Standard

maryjess committed:
Merge branch 'branch-A-CodingStandard'

maryjess committed:
Create A-Checkstyle

YH-15 commented on src/main/java/duke/Command.java:
```diff
@@ -1,5 +1,8 @@
+/**
+ * Command enumeration that lists the type of command the user inputs in the command line.
+ */
 public enum Command {
```
I love your design of the Command enum class!

YH-15 commented on src/main/java/Duke.java:
```diff
@@ -1,18 +1,27 @@
+/**
+ * Main class Duke that runs the application.
+ */
 public class Duke {
```
Very enlightening to see such a consistent level of abstraction within Duke class.

YH-15 commented on src/main/java/Duke.java:
```diff
@@ -44,6 +44,19 @@ public static void main(String[] args) {
+                    System.out.println(divider + indent + "Error: " + exception + "\n" + divider);
+                }
             } else {
                 boolean response = false;
```
It would be great if the boolean variable could be named so that it sounds like a boolean.

YH-15 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,66 @@
+            }
+        }
+        return result;
+    }
```
Impressive exception handling here.

YH-15 reviewed (commented):
The code is very well designed. It looks nice in terms of the coding standard.
Many possible exceptions are handled, and the level of abstraction within fragments is quite consistent.

maryjess committed:
Integrate GUI

snigloo commented on src/main/java/duke/Parser.java:
```diff
@@ -9,11 +9,14 @@
+     * @param input Input of user
+     * @return Type of input
      */
     public static Command parse(String input) {
```
Great use of the parser class to abstract parsing from the main Duke class!

snigloo commented on src/main/java/duke/TaskList.java:
```diff
@@ -49,45 +63,50 @@ public ArrayList<String> extractToStringArray() {
-     * @param task Task object
-     * @return boolean response
+     * Adds Task to the current task list.
+     * @param task Task to be added
```
Remember to put punctuation after every parameter of your JavaDocs. :)

snigloo commented on src/test/java/duke/DeadlineTest.java:
```diff
@@ -24,8 +27,8 @@ void getRawDateTimeTest() {
         LocalDateTime today = LocalDateTime.now();
```
Test methods could be named as dateTimeTest_RawDateTime (following `featureUnderTest_testScenario`).

snigloo commented on src/test/java/duke/ParserTest.java:
```diff
@@ -20,15 +21,15 @@ void parseTest() {
         String command7 = "list";
```
Great thorough testing! 🥳

snigloo reviewed (commented):
Overall, it has been a pleasure reading your code. Great use of abstraction and code is generally clear and follows the coding standards.

maryjess committed:
Additional error handling

The user may enter incomplete inputs such as 'mark', 'unmark', or
'delete' without specifying which Task they are referring to. Other
example includes creating Task without specifying the description.

Let's catch these errors and improve UX performance as the error
statement is only printed in the terminal, while the GUI do not show
anything.

maryjess committed:
Add exceptions

One of the most common errors that may be produced from this
application is parsing error. For example, if the user did not
enter the date and time for a Deadline and Event according to our
format.

Let's introduce a new class of Exception called MewException
to handle all possible exceptions more specifically.

maryjess committed:
Add Assertions

It is important to make sure the code runs correctly at any moment.

To support this, let's add assertions to some parts of the code.

maryjess committed:
Add Assertions

It is important to make sure the code runs correctly at any moment.

To support this, let's add assertions to some parts of the code.

maryjess committed:
Improve code quality: variable naming and JavaDoc

It is important to maintain a good code quality by continously
ensuring a good variable name and documentation of the code.

Let's use appropriate variable naming and appropriate explanation
for JavaDoc documentation.

maryjess committed:
Improve code quality: variable naming and JavaDoc

It is important to maintain a good code quality by continously ensuring a good variable name and documentation of the code.

Let's use appropriate variable naming and appropriate explanation for JavaDoc documentation.

maryjess committed:
Add CI workflow

Continous Integration (CI) as one of the workflows added is
important to ensure that processes such as 'run' and 'build' run
smoothly, before creating Pull Request or after pushing to GitHub.

Let's add CI workflow.

maryjess committed:
Update CI workflow configuration

maryjess committed:
Add CI workflow

Continous Integration (CI) as one of the workflows added is important to ensure that processes such as 'run' and 'build' run smoothly, before creating Pull Request or after pushing to GitHub.

Let's add CI workflow.

maryjess committed:
Add C-Update feature

It will be more convenient if the user can update the description
of an already created Task without deleting it and creating a new
one.

Let's integrate update description feature to this application.

maryjess committed:
Fix checkstyle error and update test class

maryjess committed:
Add C-Update feature

It will be more convenient if the user can update the description of an already created Task without deleting it and creating a new one.

Let's integrate update description feature to this application.

maryjess committed:
Refactor input-processing method processResponse

maryjess committed:
Fix storage issue

maryjess committed:
Refactor code and fix bug

maryjess committed:
Style send button

maryjess committed:
Fix checkstyle

maryjess committed:
Add user guide

maryjess committed:
Style send button and improve GUI

maryjess committed:
Create user guide

maryjess committed:
Hot fix user guide

maryjess committed:
Add UI image

maryjess committed:
Hot fix user guide

maryjess committed:
Refactor UI image

maryjess committed:
Update JavaFX version

maryjess committed:
Personalise UI

maryjess committed:
Personalise UI

maryjess committed:
Add user dialog for end input

maryjess committed:
Handle empty input

maryjess committed:
Add help feature

maryjess committed:
Handle parsing error

maryjess committed:
Handle separator error from input

maryjess committed:
Handle more exceptions

maryjess committed:
Update UI picture

maryjess committed:
Update README
